Chatkit/Model/User.swift

```diff
@@ -0,0 +1,68 @@
+import Foundation
+import CoreData
+
+public struct User {
+    
+    // MARK: - Properties
+    
+    public let identifier: String
+    public let name: String?
+    public let avatar: URL?
+    public let presenceState: PresenceState
+    public let metadata: Metadata?
+    public let createdAt: Date
+    public let updatedAt: Date
+    
+    let objectID: NSManagedObjectID
+    
+    // MARK: - Initializers
+    
+    init(identifier: String, name: String?, avatar: URL?, presenceState: PresenceState, metadata: Metadata?, createdAt: Date, updatedAt: Date, objectID: NSManagedObjectID) {
+        self.identifier = identifier
+        self.name = name
+        self.avatar = avatar
+        self.presenceState = presenceState
+        self.metadata = metadata
+        self.createdAt = createdAt
+        self.updatedAt = updatedAt
+        self.objectID = objectID
+    }
+    
+}
+
+// MARK: - Hashable
+
+extension User: Hashable {
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(self.identifier)
+    }
+    
+}
+
+// MARK: - Equatable
+
+extension User: Equatable {
+    
+    public static func == (lhs: User, rhs: User) -> Bool {
```

Comment:
It's something for the future but we could use [Sourcery](https://www.raywenderlich.com/501-sourcery-tutorial-generating-swift-code-for-ios) to auto-generate these things.

Reply:
Sure, that an option for us to consider.